src/Compilers/CSharp/Test/Semantic/SourceGeneration/StateTableTests.cs

```diff
@@ -306,6 +361,17 @@ private void AssertTableEntries<T>(NodeStateTable<T> table, IList<(T item, Entry
             }
         }
 
+        private void AssertTableEntries<T>(NodeStateTable<ImmutableArray<T>> table, IList<(ImmutableArray<T> item, EntryState state)> expected)
+        {
+            int index = 0;
+            foreach (var entry in table)
+            {
+                Assert.Equal(expected[index].item, (IEnumerable<T>)entry.item);
```

Comment:
Consider using the `AssertEx` helpers that do nice output diffing to make failures more diagnosable.